embassy-mcxa/src/lpuart/mod.rs

```diff
@@ -692,10 +696,37 @@ pub struct LpuartRx<'a, M: Mode> {
     info: Info,
     _rx_pin: Peri<'a, AnyPin>,
     _rts_pin: Option<Peri<'a, AnyPin>>,
-    _rx_dma: Option<Channel<'a>>,
     mode: PhantomData<(&'a (), M)>,
 }
 
+/// Lpuart TX driver with DMA support.
+pub struct LpuartTxDma<'a, T: Instance, C: DmaChannelTrait> {
+    info: Info,
+    _tx_pin: Peri<'a, AnyPin>,
+    tx_dma: DmaChannel<C>,
```

Comment:
Shouldn't this be something like a `Peri<'a, AnyChannel>`? That is, be a peripheral and a type-erased DMA channel type?

Reply:
I think it's reasonable, DmaChannel is also a zst, and has useful helper methods.